api/basic/add-pin.ts

```diff
@@ -0,0 +1,178 @@
+import { DynamoDBClient, ConditionalCheckFailedException } from '@aws-sdk/client-dynamodb'
+import { DynamoDBDocumentClient, GetCommand, PutCommand } from '@aws-sdk/lib-dynamodb'
+import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs'
+import { APIGatewayProxyEventV2 } from 'aws-lambda'
+import { ClusterAddResponse, Pin, Response } from './schema.js'
+import { CID } from 'multiformats/cid'
+import { Multiaddr } from 'multiaddr'
+import { nanoid } from 'nanoid'
+
+interface UpsertPinInput {
+  cid: string
+  dynamo: DynamoDBClient
+  table: string
+}
+
+interface AddToQueueInput {
+  cid: string
+  origins: string[]
+  bucket: string
+  sqs: SQSClient
+  queueUrl: string
+}
+
+// type AddPinInput = AddToQueueInput & UpsertPinInput
+interface AddPinInput extends UpsertPinInput, AddToQueueInput {}
+
+/**
+ * AWS API Gateway handler for POST /pin/${cid}?&origins=${multiaddr},${multiaddr}
+ * Collect the params and delegate to addPin to do the work
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ */
+export async function handler (event: APIGatewayProxyEventV2): Promise<Response> {
+  const {
+    TABLE_NAME: table = '',
+    BUCKET_NAME: bucket = '',
+    QUEUE_URL: queueUrl = '',
+    CLUSTER_BASIC_AUTH_TOKEN: token = '',
+    // set for testing
```

Comment:
any other way of not setting this here for testing?

Reply:
I cant't change the lambda handler function signature to pass in an sqs or dynamo client, but to test it locally I need to use a client that points to the local endpoint. I don't think there is another way... but my hot brain might be missing something.